include/swift/SILOptimizer/Utils/Existential.h

```diff
@@ -86,11 +89,22 @@ struct ConcreteExistentialInfo {
   // Constructs a valid ConcreteExistentialInfo object if successfull.
   ConcreteExistentialInfo(Operand &openedUse);
 
+  // This constructor uses protocol conformance analysis to determine concrete
+  // type of the existential. It has the side effect of modifying the body of
+  // the function for which we are propagating concrete type, thus a builder is
+  // passed.
+  ConcreteExistentialInfo(ProtocolConformanceAnalysis *PCA,
+                          ClassHierarchyAnalysis *CHA, SILBuilder &Builder,
+                          ProtocolDecl *Protocol, FullApplySite Apply,
+                          SILValue &Arg, Operand &ArgOperand);
+
   ConcreteExistentialInfo(ConcreteExistentialInfo &) = delete;
 
+  /// We do not not need to check for InitExistential not being not null
```

Comment:
I don't understand the comment. Aren't you allowing InitExistential to be null when the concrete type was derived through analysis instead?

Reply:
Will rephrase the comment.